src/domain/fido-credential.ts

```diff
@@ -0,0 +1,290 @@
+          }
+          Luckily, to do that, we just need to prefix it with constant 26 bytes (metadata is constant).
+      */
+
+    pkBuffer = Buffer.concat([
+      Buffer.from('3059301306072a8648ce3d020106082a8648ce3d030107034200', 'hex'),
+      pkBuffer
+    ])
+
+    type = 'PUBLIC KEY'
+  } else {
+    type = 'CERTIFICATE'
+  }
+
+  const b64cert = pkBuffer.toString('base64')
+
+  let PEMKey = ''
+  for (let i = 0; i < Math.ceil(b64cert.length / 64); i++) {
+    const start = 64 * i
+
+    PEMKey += b64cert.substr(start, 64) + '\n'
+  }
+
+  PEMKey = `-----BEGIN ${type}-----\n` + PEMKey + `-----END ${type}-----\n`
+
+  return PEMKey
+}
+
+/**
+ * Takes COSE encoded public key and converts it to RAW PKCS ECDHA key
+ * @param  {Buffer} COSEPublicKey - COSE encoded public key
+ * @return {Buffer}               - RAW PKCS encoded public key
+ */
+export function COSEECDHAtoPKCS (COSEPublicKey: Buffer): Buffer {
+  /*
+     +------+-------+-------+---------+----------------------------------+
+     | name | key   | label | type    | description                      |
+     |      | type  |       |         |                                  |
+     +------+-------+-------+---------+----------------------------------+
+     | crv  | 2     | -1    | int /   | EC Curve identifier - Taken from |
+     |      |       |       | tstr    | the COSE Curves registry         |
+     |      |       |       |         |                                  |
+     | x    | 2     | -2    | bstr    | X Coordinate                     |
+     |      |       |       |         |                                  |
+     | y    | 2     | -3    | bstr /  | Y Coordinate                     |
+     |      |       |       | bool    |                                  |
+     |      |       |       |         |                                  |
+     | d    | 2     | -4    | bstr    | Private key                      |
+     +------+-------+-------+---------+----------------------------------+
+  */
+
+  const coseStruct = decodeAllSync(COSEPublicKey)[0]
+  const tag = Buffer.from([0x04])
+  const x = coseStruct.get(-2)
+  const y = coseStruct.get(-3)
+
+  return Buffer.concat([tag, x, y])
+}
+
+/**
+ * Takes signature, data and PEM public key and tries to verify signature
+ * @param  {Buffer} signature
+ * @param  {Buffer} data
+ * @param  {String} publicKey - PEM encoded public key
+ * @return {Boolean}
+ */
+export function verifySignature (signature: Buffer, data: Buffer, publicKey: string): boolean {
+  return createVerify('SHA256')
+    .update(data)
+    .verify(publicKey, signature)
+}
+
+export async function validateAttestation (
+  attestationObject: string,
+  clientDataJSON: string
+): Promise<boolean> {
+  const attestation = await unpackAttestationObject(attestationObject)
+  const { attStmt, authData } = attestation
+
+  const auth = parseRegCredAuthData(Buffer.from(authData))
+
+  // user must be presented!
+  if (!(auth.flags & U2F_USER_PRESENTED)) {
```

Comment:
Did you mean `&&` here? Or bitwise `AND`?

Reply:
it is a bit flag so bitwise AND

Reply:
it must be bitwise